python/GafferTest/DotTest.py

```diff
@@ -165,5 +166,28 @@ def testArrayPlug( self ) :
 		self.assertTrue( d["out"][0].source().isSame( i ) )
 		self.assertTrue( d["out"][1].source().isSame( n1["a"][1] ) )
 
+	def testSetupCopiesPlugColorMetadata( self ):
+
+		s = Gaffer.ScriptNode()
+
+		s["n1"] = GafferTest.AddNode()
+		s["d"] = Gaffer.Dot()
+
+		plug = s["n1"]["op1"]
+
+		connectionColor = IECore.Color3fData( IECore.Color3f( 0.1 , 0.2 , 0.3 ) )
```

Comment:
This whole "wrap the thing in Data and then use .value to get at it" thing is pretty pervasive in Cortex, and it's a bit of a pain. In Gaffer I've tried to make it more transparent on the Python side of things, so you can simply do `connectionColor = IECore.Color3f( ... )` here, and the `Metadata.registerValue()` call below will transparently convert to Data on the way in, just at `Metadata.value()` will convert back on the way out.